integration/test_collection.py

```diff
@@ -61,18 +68,116 @@ def test_create_and_delete(client: weaviate.Client):
     assert not client.collection.exists(name)
 
 
-def test_insert(client: weaviate.Client):
-    name = "TestInsert"
+def test_create_get_and_delete_with_supported_generic(client: weaviate.Client):
+    name = "TestCreateGetAndDeleteGeneric"
     collection_config = CollectionConfig(
         name=name,
         properties=[Property(name="Name", data_type=DataType.TEXT)],
         vectorizer=Vectorizer.NONE,
     )
-    collection = client.collection.create(collection_config)
-    uuid = collection.data.insert(properties={"name": "some name"})
-    assert collection.data.get_by_id(uuid).properties["name"] == "some name"
+    client.collection.create(collection_config)
+
+    class Right(TypedDict):
+        name: str
 
+    client.collection.get(name, Right)
+    assert client.collection.exists(name)
     client.collection.delete(name)
+    assert not client.collection.exists(name)
+
+
+def test_get_with_empty_class(client: weaviate.Client):
+    class Wrong:
+        name: str
+
+    with pytest.raises(TypeError) as error:
+        client.collection.get("NotImportant", Wrong)
+    assert (
+        error.value.args[0]
+        == "data_model can only be a dict or a class that inherits from TypedDict"
+    )
+
+
+def test_get_with_dataclass(client: weaviate.Client):
+    @dataclass
+    class Wrong:
+        name: str
+
+    with pytest.raises(TypeError) as error:
+        client.collection.get(
+            "NotImportant", Wrong
+        )  # bad type because dataclass not bound by generic
+    assert (
+        error.value.args[0]
+        == "data_model can only be a dict or a class that inherits from TypedDict"
+    )
+
+
+def test_get_with_initialisable_class(client: weaviate.Client):
+    class Wrong:
+        name: str
+
+        def __init__(self, name: str):
+            self.name = name
+
+    with pytest.raises(TypeError) as error:
+        client.collection.get("NotImportant", Wrong)
+    assert (
+        error.value.args[0]
+        == "data_model can only be a dict or a class that inherits from TypedDict"
+    )
+
+
+def test_get_with_pydantic_class(client: weaviate.Client):
+    class Wrong(BaseModel):
+        name: str
+
+    with pytest.raises(TypeError) as error:
+        client.collection.get("NotImportant", Wrong)
+    assert (
+        error.value.args[0]
+        == "data_model can only be a dict or a class that inherits from TypedDict"
```

Comment:
Ah okay, perhaps just `Optional[TypedDict]` since passing `None` defaults to `Dict[str, Any]` anyway. So it would be
>If provided, `data_model` can only be a class that inherits from `TypedDict`

The thing is, they could provide `Dict[str, str]` if they know their dictionary is only made of string values but don't want to use a `TypedDict`. Seems like an niche use-case but it should be supportable nonetheless

Reply:
How's this?
>data_model can only be a dict type, e.g. Dict[str, str], or a class that inherits from TypedDict

Reply:
perfect!